library/bignum_mod.h

```diff
@@ -173,7 +183,67 @@ void mbedtls_mpi_mod_modulus_free( mbedtls_mpi_mod_modulus *m );
 /* END MERGE SLOT 6 */
 
 /* BEGIN MERGE SLOT 7 */
+/** Read a residue from a byte buffer.
+ *
+ * The residue will be automatically converted to the internal representation
+ * based on the value of `m->int_rep` field.
+ *
+ * The modulus \p m will be the modulus associated with \p r. The residue \p r
+ * should only be used in operations where the modulus is \p m or a modulus
+ * equivalent to \p m (in the sense that all their fields or memory pointed by
+ * their fields hold the same value).
+ *
+ * \param r         The address of the residue. It must have as many limbs as
+ *                  the modulus \p m.
+ * \param m         The address of the modulus.
+ * \param buf       The input buffer to import from.
+ * \param buflen    The length in bytes of \p buf.
+ * \param ext_rep   The endianness of the number in the input buffer.
+ *
+ * \return       \c 0 if successful.
+ * \return       #MBEDTLS_ERR_MPI_BUFFER_TOO_SMALL if \p r isn't
```

Comment:
The `MBEDTLS_ERR_MPI_BUFFER_TOO_SMALL` error doesn't mean that `r` is in [2^k,∞), it might just have too many leading zeroes. Read doesn't inspect the value being read, like with the write, it is the caller's responsibility to handle leading zeroes based on the sensitivity of the data.